src/transformers/modeling_deberta.py

```diff
@@ -0,0 +1,1213 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch DeBERTa model. """
+
+import copy
+import json
+import logging
+import math
+import os
+from collections import Sequence
+
+import torch
+from packaging import version
+from torch import nn
+from torch.nn import CrossEntropyLoss
+
+from .configuration_deberta import DeBERTaConfig
+from .file_utils import add_start_docstrings
+from .modeling_outputs import BaseModelOutputWithPooling, SequenceClassifierOutput
+from .modeling_utils import PreTrainedModel
+
+
+if version.Version(torch.__version__) >= version.Version("1.0.0"):
+    from torch import _softmax_backward_data as _softmax_backward_data
+else:
+    from torch import softmax_backward_data as _softmax_backward_data
+
+__all__ = [
+    "DeBERTaModel",
+    "DeBERTaForSequenceClassification",
+    "DeBERTaPreTrainedModel",
+    "DEBERTA_PRETRAINED_MODEL_ARCHIVE_LIST",
+]
+
+logger = logging.getLogger(__name__)
+
+####################################################
+# This list contrains shortcut names for some of
+# the pretrained weights provided with the models
+####################################################
+DEBERTA_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "microsoft/deberta-base",
+    "microsoft/deberta-large",
+]
+
+
+def gelu(x):
+    """Implementation of the gelu activation function.
+    For information: OpenAI GPT's gelu is slightly different (and gives slightly different results):
+    0.5 * x * (1 + torch.tanh(math.sqrt(2 / math.pi) * (x + 0.044715 * torch.pow(x, 3))))
+    """
+    return x * 0.5 * (1.0 + torch.erf(x / math.sqrt(2.0)))
+
+
+def swish(x):
+    return x * torch.sigmoid(x)
+
+
+def linear_act(x):
+    return x
+
+
+ACT2FN = {
+    "gelu": gelu,
+    "relu": torch.nn.functional.relu,
+    "swish": swish,
+    "tanh": torch.nn.functional.tanh,
+    "linear": linear_act,
+    "sigmoid": torch.sigmoid,
+}
+
+
+def traceable(cls):
+    """Decorator over customer functions
+    There is an issue for tracing customer python torch Function, using this decorator to work around it.
+    e.g.
+    @traceable
+    class MyOp(torch.autograd.Function):
+    xxx
+    """
+
+    class _Function(object):
+        @staticmethod
+        def apply(*args):
+            jit_trace = os.getenv("JIT_TRACE", "False").lower() == "true"
+            if jit_trace:
+                return cls.forward(_Function, *args)
+            else:
+                return cls.apply(*args)
+
+        @staticmethod
+        def save_for_backward(*args):
+            pass
+
+    _Function.__name__ = cls.__name__
+    _Function.__doc__ = cls.__doc__
+    return _Function
+
+
+class AbsModelConfig(object):
```

Comment:
Could you add a docstring to explain what this is for? Why not use the `DeBERTaConfiguration` object as Sam mentioned?

Reply:
Could you add a docstring to explain what this is for? Why not use the DeBERTaConfiguration object as Sam mentioned?